aten/src/ATen/native/cuda/IntegerMatmul.cu

```diff
@@ -0,0 +1,77 @@
+#include <ATen/cuda/CUDABlas.h>
+#include <ATen/cuda/CUDAApplyUtils.cuh>
+#include <ATen/cuda/detail/KernelUtils.h>
+#include <ATen/native/cuda/Loops.cuh>
+#include <c10/macros/Macros.h>
+
+namespace at {
+namespace cuda {
+namespace blas {
+
+  template <typename scalar_t, bool transA=false, bool transB=false>
+  __global__ void gemm_kernel(int64_t M, int64_t N, int64_t K, scalar_t alpha,
+                                                  const scalar_t* a, int64_t lda, const scalar_t * b, int64_t ldb,
```

Comment:
Making `a` and `b` `const scalar_t* __restrict__` should hopefully improve the performance, as `a` and  `b` would be saved into the L1 cache. This should be good as there quite a big number of repeated accesses in the kernel.

Reply:
Done.